container/lxd/cluster_test.go

```diff
@@ -0,0 +1,30 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package lxd_test
+
+import (
+	"github.com/golang/mock/gomock"
+	gc "gopkg.in/check.v1"
+
+	"github.com/juju/juju/container/lxd"
+	lxdtesting "github.com/juju/juju/container/lxd/testing"
+)
+
+type clusterSuite struct {
+	lxdtesting.BaseSuite
+}
+
+var _ = gc.Suite(&clusterSuite{})
+
+func (s *imageSuite) TestUseTarget(c *gc.C) {
+	ctrl := gomock.NewController(c)
+	defer ctrl.Finish()
+
+	c1Svr := s.NewMockServerClustered(ctrl, "cluster-1")
+	c2Svr := s.NewMockServerClustered(ctrl, "cluster-2")
+
+	c1Svr.EXPECT().UseTarget("cluster-2").Return(c2Svr)
+
+	lxd.NewServer(c1Svr).UseTargetServer("cluster-2")
```

Comment:
Should there be a test to show what happens if someone attempts to use a server that isn't a valid cluster name?

```go
lxd.NewServer(c1Svr).UseTargetServer("cluster-bad")
```

Reply:
I have added such a test.

As per the edited PR description, I had to add an error to the return from ```NewServer```, so there are a few call sites in the diff now.